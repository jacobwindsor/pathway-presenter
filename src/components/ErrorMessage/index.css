.error-wrapper {
    position: relative;
    width: 100%;
    height: 100%;
}

.error-wrapper .error-message {
    text-align: center;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 1.6rem;
}