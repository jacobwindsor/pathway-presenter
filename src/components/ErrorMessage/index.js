import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage =  (props) => <p>Error: {props.message}</p>;

ErrorMessage.propTypes = {
    message: PropTypes.string.isRequired
};

export default ErrorMessage;