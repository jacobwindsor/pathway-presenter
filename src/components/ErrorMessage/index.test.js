import React from 'react';
import { shallow } from 'enzyme';
import ErrorMessage from './index';
import toJson from 'enzyme-to-json';

it('renders without crashing', () => {
  const comp = shallow(<ErrorMessage message="Some error" />);
  expect(toJson(comp)).toMatchSnapshot();
});

it('Shows the correct error message', () => {
  const comp = shallow(<ErrorMessage message="My custom error" />);
  expect(comp.contains('My custom error')).toBeTruthy();
});
