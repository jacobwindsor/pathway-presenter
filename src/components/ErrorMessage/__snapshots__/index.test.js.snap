// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<MuiThemeProvider>
  <div
    className="error-wrapper"
  >
    <Paper
      circle={false}
      className="error-message"
      rounded={true}
      transitionEnabled={true}
      zDepth={1}
    >
      <img
        alt="Error exclamation mark"
        src="error.svg"
      />
      <h1>
        Woops
      </h1>
      <h3>
        Some error
      </h3>
    </Paper>
  </div>
</MuiThemeProvider>
`;
